dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -783,12 +784,13 @@ public Workflow publish(@ApiParam(hidden = true) @Parameter(hidden = true, name
         @ApiParam(value = "Filter, this is a search string that filters the results.") @DefaultValue("") @QueryParam("filter") String filter,
         @ApiParam(value = "Sort column") @DefaultValue("stars") @QueryParam("sortCol") String sortCol,
         @ApiParam(value = "Sort order", allowableValues = "asc,desc") @DefaultValue("desc") @QueryParam("sortOrder") String sortOrder,
-        @ApiParam(value = "services", defaultValue = "false") @DefaultValue("false") @QueryParam("services") boolean services,
+        @ApiParam(value = "Should only be used by Dockstore CLI versions < 1.12.0. Indicates whether to get a service or workflow") @DefaultValue("false") @QueryParam("services") boolean services,
+        @ApiParam(value = "Which workflow subclass to retrieve. If present takes precedence over services parameter", defaultValue = "") @QueryParam("subclass") WorkflowSubClass subclass,
```

Comment:
It will work the way it is, but I should probably remove the defaultValue, it's not adding anything.

Because the type of the parameter is `WorkflowSubClass`, Jersey will deserialize it into a `WorkflowSubClass` or `null`. An empty string gets deserialized into `null`. 

BTW, an invalid value causes a bad request:

```
curl localhost:8080/workflows/published?subclass=abc
{"code":400,"message":"query param subclass must be one of [BIOWORKFLOW, SERVICE, APPTOOL]"}
```

Removed defaultValue in 117857de47b88ea961e8f7da9381c4f7a9bc5494

Reply:
I interpreted this code wrong.  Charles is correct, there was no problem!